source/DataBase/QueryOperationBuilder.cpp

```diff
@@ -0,0 +1,129 @@
+
+#include "QueryOperationBuilder.hpp"
+
+#include <format>
+
+#include "common/DataBaseCommon.hpp"
+namespace dataBase {
+namespace {
+std::unordered_map<common::DataBaseRequest::TableType, std::string> tableNames =
+    {{common::DataBaseRequest::TableType::Users, common::users},
+     {common::DataBaseRequest::TableType::Passwords, common::passwords},
+     {common::DataBaseRequest::TableType::PasswordHistory,
+      common::passwordHistory},
+     {common::DataBaseRequest::TableType::Categories, common::categories}};
+
+std::string getTableName(common::DataBaseRequest::TableType table) {
+  if (auto it = tableNames.find(table); it != tableNames.end())
+    return it->second;
+
+  return {};
+}
+
+bool isDataEmpty(const common::RequestData &data) {
+  return data.empty() ? true : false;
+}
+
+bool isPrimaryKeyEmpty(const common::PrimaryKey &primaryKey) {
+  return (primaryKey.first.empty() || primaryKey.second.empty()) ? true : false;
+}
+}  // namespace
+QueryOperationBuilder::QueryOperationBuilder() {
+  queryBuilders_ = {{common::DataBaseRequest::OperationType::Insert,
+                     std::bind(&QueryOperationBuilder::buildInsertQuery, this,
+                               std::placeholders::_1, std::placeholders::_2)},
+                    {common::DataBaseRequest::OperationType::Delete,
+                     std::bind(&QueryOperationBuilder::buildDeleteQuery, this,
+                               std::placeholders::_1, std::placeholders::_2)},
+                    {common::DataBaseRequest::OperationType::Update,
+                     std::bind(&QueryOperationBuilder::buildUpdateQuery, this,
+                               std::placeholders::_1, std::placeholders::_2)},
+                    {common::DataBaseRequest::OperationType::Select,
+                     std::bind(&QueryOperationBuilder::buildSelectQuery, this,
+                               std::placeholders::_1, std::placeholders::_2)}};
+}
```

Comment:
It is unnecessary to use `std::bind` here and can be replaced with lambda functions, which are more readable and efficient.

```c++
QueryOperationBuilder::QueryOperationBuilder() {
  queryBuilders_ = {{common::DataBaseRequest::OperationType::Insert,
                     [this](const common::DataBaseRequest &requestData,
                            const std::string &tableName) {
                       return this->buildInsertQuery(requestData, tableName);
                     }},
                    {common::DataBaseRequest::OperationType::Delete,
                     [this](const common::DataBaseRequest &requestData,
                            const std::string &tableName) {
                       return this->buildDeleteQuery(requestData, tableName);
                     }},
                    {common::DataBaseRequest::OperationType::Update,
                     [this](const common::DataBaseRequest &requestData,
                            const std::string &tableName) {
                       return this->buildUpdateQuery(requestData, tableName);
                     }},
                    {common::DataBaseRequest::OperationType::Select,
                     [this](const common::DataBaseRequest &requestData,
                            const std::string &tableName) {
                       return this->buildSelectQuery(requestData, tableName);
                     }}};
}
```